internal/dataplane/translator/translate_ingress.go

```diff
@@ -81,53 +84,133 @@ func (t *Translator) ingressRulesFromIngressV1() ingressRules {
+	storer store.Storer,
+	failuresCollector *failures.ResourceFailuresCollector,
+	allDefaultBackends []netv1.Ingress,
+	features FeatureFlags,
+) (kongstate.Service, bool) {
+	// Sort the default backends by creation timestamp, so that the oldest one is picked.
 	sort.SliceStable(allDefaultBackends, func(i, j int) bool {
 		return allDefaultBackends[i].CreationTimestamp.Before(&allDefaultBackends[j].CreationTimestamp)
 	})
 
 	if len(allDefaultBackends) > 0 {
 		ingress := allDefaultBackends[0]
-		defaultBackend := allDefaultBackends[0].Spec.DefaultBackend
-		port := subtranslator.PortDefFromServiceBackendPort(&defaultBackend.Service.Port)
-		serviceName := fmt.Sprintf(
-			"%s.%s.%s",
-			allDefaultBackends[0].Namespace,
-			defaultBackend.Service.Name,
-			port.CanonicalString(),
-		)
-		service := kongstate.Service{
-			Service: kong.Service{
-				Name: kong.String(serviceName),
-				Host: kong.String(fmt.Sprintf(
-					"%s.%s.%s.svc",
-					defaultBackend.Service.Name,
-					ingress.Namespace,
-					port.CanonicalString(),
-				)),
-				Port:           kong.Int(DefaultHTTPPort),
-				Protocol:       kong.String("http"),
-				ConnectTimeout: kong.Int(DefaultServiceTimeout),
-				ReadTimeout:    kong.Int(DefaultServiceTimeout),
-				WriteTimeout:   kong.Int(DefaultServiceTimeout),
-				Retries:        kong.Int(DefaultRetries),
-				Tags:           util.GenerateTagsForObject(&ingress),
-			},
-			Namespace: ingress.Namespace,
-			Backends: []kongstate.ServiceBackend{{
-				Name:    defaultBackend.Service.Name,
-				PortDef: port,
-			}},
-			Parent: &ingress,
+		defaultBackend := ingress.Spec.DefaultBackend
+		route := translateIngressDefaultBackendRoute(&ingress, util.GenerateTagsForObject(&ingress), features.ExpressionRoutes)
+
+		// If the default backend is defined as an arbitrary resource, then we need handle it differently.
+		if resource := defaultBackend.Resource; resource != nil {
+			return translateIngressDefaultBackendResource(
+				resource,
+				ingress,
+				route,
+				storer,
+				failuresCollector,
+				features,
+			)
 		}
-		r := translateIngressDefaultBackendRoute(&ingress, util.GenerateTagsForObject(&ingress), expressionRoutes)
-		service.Routes = append(service.Routes, *r)
-		return service, true
+
+		// Otherwise, the default backend is defined as a Kubernetes Service.
+		return translateIngressDefaultBackendService(ingress, route)
 	}
 
 	return kongstate.Service{}, false
 }
 
+func translateIngressDefaultBackendResource(
+	resource *corev1.TypedLocalObjectReference,
+	ingress netv1.Ingress,
+	route *kongstate.Route,
+	storer store.Storer,
+	failuresCollector *failures.ResourceFailuresCollector,
+	features FeatureFlags,
+) (kongstate.Service, bool) {
+	if !subtranslator.IsKongServiceFacade(resource) {
+		gk := resource.Kind
+		if resource.APIGroup != nil {
+			gk = *resource.APIGroup + "/" + gk
+		}
+		failuresCollector.PushResourceFailure(fmt.Sprintf("default backend: unknown resource type %s", gk), &ingress)
```

Comment:
```suggestion
		failuresCollector.PushResourceFailure(fmt.Sprintf("default backend: unsupported resource type %s", gk), &ingress)
```
I think resources other than `KongServiceFacade` is not an "unknown" type, but an "unsupported" type.